source/sqlsrv/stmt.cpp

```diff
@@ -1974,9 +1977,11 @@ void parse_param_array( _Inout_ ss_sqlsrv_stmt* stmt, _Inout_ zval* param_array,
 
         sql_type = sqlsrv_sql_type.typeinfo.type;
     }
-    // else the sql type and size are uknown, so tell the core layer to use its defaults
+    // else the sql type and size are unknown, so tell the core layer to use its defaults
     else {
-
+        CHECK_CUSTOM_ERROR( !stmt->prepared && stmt->conn->ce_option.enabled, stmt, SS_SQLSRV_ERROR_AE_QUERY_SQLTYPE_REQUIRED ) {
+            throw ss::SSException();
+        }
```

Comment:
So only throw this error when the user is not using prepare() for AE?
What if the user uses the param array but does not specify SQL type?

Reply:
yes, when user uses prepare() there is no problem since SQLDescribeParam is called after SQLPrepare. When query is used however, SQLDescribeParam is call BEFORE SQLExecDirect, resulting a function sequence error.

The part that user uses a param array but does not specify SQL type is handled in line 1225 to 1227.